app/src/main/java/com/asfoundation/wallet/util/UrlParser.kt

```diff
@@ -0,0 +1,31 @@
+package com.asfoundation.wallet.util
+
+import java.io.UnsupportedEncodingException
+import java.net.URLDecoder
+
+class UrlUtmParser {
+
+  companion object {
+
+    @Throws(UnsupportedEncodingException::class)
+    fun splitQuery(urlParams: String): Map<String, MutableList<String?>>? {
+      val queryPairs: MutableMap<String, MutableList<String?>> = LinkedHashMap()
+      val pairs: List<String> = urlParams.split("&")
+      for (pair in pairs) {
+        val idx = pair.indexOf("=")
```

Comment:
Why not simply use `split` here and remove all those calculations involving index and mutable lists update? 🤔